torch/_refs/linalg/__init__.py

```diff
@@ -0,0 +1,103 @@
+import torch
+
+import torch._prims as prims
+import torch._refs as refs
+from torch._prims.wrappers import out_wrapper
+
+from torch._prims.utils import (
+    DimsType,
+    TensorLikeType,
+    NumberType,
+    corresponding_real_dtype,
+    is_float_dtype,
+    get_computation_dtype,
+    reduction_dtypes,
+    REDUCTION_OUTPUT_TYPE_KIND,
+)
+import torch._refs.linalg as linalg
+import torch._refs.linalg.utils
+
+from typing import Optional, List
+from functools import partial
+
+__all__ = [
+    "vector_norm",
+]
+
+
+@out_wrapper
+def vector_norm(
+    x: TensorLikeType,
+    ord: float = 2.0,
+    dim: Optional[DimsType] = None,
+    keepdim: bool = False,
+    *,
+    dtype: Optional[torch.dtype] = None,
+):
+    # Checks
+    linalg.utils.check_fp_or_complex(x, "linalg.vector_norm", half=True)
+
+    if isinstance(dim, int):
+        dim = [dim]  # type: ignore[assignment]
+    elif not isinstance(dim, List) and dim is not None:
+        # refs.sum just accepts List rather than DimType
+        dim = list(dim)  # type: ignore[assignment]
+
+    # TODO These things are TORCH_CHECKS, replace them with check(a,b)
+    # once https://github.com/pytorch/pytorch/pull/78014 is merged
+    if x.numel() == 0 and (ord < 0.0 or ord == float("inf")):
+        if dim is None or len(dim) == 0:
+            raise RuntimeError(
+                "linalg.vector_norm cannot compute the {ord} norm on an empty tensor "
+                "because the operation does not have an identity"
+            )
+        else:
+            for d in dim:
+                if x.size(d) == 0:
```

Comment:
cc @mruberry, can we call tensor methods in refs? That doesn't seem right

Reply:
Fixed